app/renderer/src/actions/Inspector.js

```diff
@@ -849,22 +901,79 @@ export function setAwaitingMjpegStream(isAwaiting) {
   };
 }
 
+export function setGestureUploadErrors(errors) {
+  return (dispatch) => {
+    dispatch({type: SET_GESTURE_UPLOAD_ERROR, errors});
+  };
+}
+
+export function uploadGesturesFromFile(fileList) {
+  return async (dispatch) => {
+    const gestures = await readTextFromUploadedFiles(fileList);
+    const invalidGestures = {};
+    const parsedGestures = [];
+    for (const gesture of gestures) {
+      const {fileName, content, error} = gesture;
+      try {
+        // Some error occured while reading the uploaded file
+        if (error) {
+          invalidGestures[fileName] = [i18n.t('gestureInvalidJsonError')];
+          return;
+        }
+        const gesture = JSON.parse(content);
+        if (!gesture.name) {
+          invalidGestures[fileName] = [i18n.t('gestureNameCannotBeEmptyError')];
+          return;
+        }
+        const actionErrors = gesture.actions
+          .map(checkErrorsInAction)
+          .reduce((acc, error) => (error.length ? acc.concat(error) : acc), []);
+
+        if (actionErrors.length) {
+          invalidGestures[fileName] = actionErrors;
+        } else {
```

Comment:
Would suggest maybe aligning this to the previous checks, by adding a `continue` and moving the `else` branch contents one indent earlier

Reply:
Makes sense. Will keep it consistent.